beacon_node/eth2-libp2p/src/types/pubsub.rs

```diff
@@ -66,55 +68,71 @@ impl<T: EthSpec> PubsubMessage<T> {
                     continue;
                 }
                 Ok(gossip_topic) => {
-                    match gossip_topic.encoding() {
+                    let mut decompressed_data: Vec<u8> = Vec::new();
```

Comment:
Is there a reason this is isn't localised to the scope of SSZSnappy?

Reply:
The aim was to potentially save an allocation if an SSZ encoding was received by returning a reference in both match cases. Can't return a reference to the vec if it's in the local scope of SSZSnappy as it goes out of scope after match statement.